src/context/GlobalContext.ts

```diff
@@ -73,20 +72,23 @@ export class GlobalContext extends Context<ClientContext> implements IGlobalCont
     getClientUrl(): string {
         return this.xrmContext.getClientUrl();
     }
+
     getCrmVersion(): string {
         return this.xrmContext.getVersion();
     }
+
     getCurrentAppUrl(): string {
         return this.xrmContext.getCurrentAppUrl
             ? this.xrmContext.getCurrentAppUrl()
             : this.xrmContext.getClientUrl();
     }
+
     getOrgUniqueName(): string {
         return this.xrmContext.getOrgUniqueName();
     }
+
     getOrganization(): IOrganization {
         const orgSettings: Xrm.OrganizationSettings = this.xrmContext.organizationSettings || {} as any;
-        orgSettings.uniqueName = this.getOrgUniqueName();

```

Comment:
Not sure. This interface comes from @types/xrm. These properties should have been defined as _readonly_ there. The only thing we could do to prevent this is to create our own version of these interfaces, by creating it ourselves or forking it from @types/xrm.